ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/messagelistpage/MessageListContainerFragmentTest.java

```diff
@@ -0,0 +1,403 @@
+  }
+
+  @Test
+  public void testOnActivityResultWithViewAttachment() {
+    // Arrange
+    int viewAttachmentCode = Whitebox.getInternalState(MessageListContainerFragment.class, "REQUEST_CODE_VIEW_ATTACHMENT_BEFORE_SENDING");
+    int resultCode = Whitebox.getInternalState(KayakoAttachmentPreviewActivity.class, "RESULT_EXIT");
+
+    // Act
+    messageListContainerFragment.onActivityResult(viewAttachmentCode, resultCode, intent);
+
+    // Assert
+    assertNull(Whitebox.getInternalState(messageListContainerFragment, "mLastFileAttachmentAttached"));
+  }
+
+  @Test
+  public void testOnDestroyView() {
+    // Arrange
+    MessageListContainerContract.Presenter presenterMock = PowerMockito.mock(MessageListContainerContract.Presenter.class);
+    Whitebox.setInternalState(messageListContainerFragment, "mPresenter", presenterMock);
+
+    // Act
+    messageListContainerFragment.onDestroyView();
+
+    // Assert
+    Mockito.verify(messageListContainerFragment).onDestroyView();
+  }
+
+  @Test
+  public void testHasPageLoaded() {
+    // Arrange
+    boolean isAdded = messageListContainerFragment.isAdded();
+    Activity activity = messageListContainerFragment.getActivity();
+
+    // Act
+    boolean pageLoaded = messageListContainerFragment.hasPageLoaded();
+
+    // Assert
+    Mockito.verify(messageListContainerFragment, Mockito.times(2)).isAdded();
+    if (isAdded) {
+      Mockito.verify(messageListContainerFragment, Mockito.times(2)).getActivity();
+    }
+    assertThat(pageLoaded, is(isAdded && activity != null));
+  }
+
+  @Test
+  public void testShowToastMessageWithResId() throws Exception {
+    // Arrange
+    Toast toastMock = Mockito.mock(Toast.class);
+    PowerMockito.when(Toast.class, "makeText", context, R.string.ko__action_contact, Toast.LENGTH_SHORT)
+        .thenReturn(toastMock);
+
+    // Act
+    messageListContainerFragment.showToastMessage(R.string.ko__action_contact);
+
+    // Assert
+    Mockito.verify(toastMock).show();
+  }
+
+  @Test
+  public void testShowToastMessageWithString() throws Exception {
+    // Arrange
+    Toast toastMock = Mockito.mock(Toast.class);
+    PowerMockito.when(Toast.class, "makeText", context, TOAST_MESSAGE, Toast.LENGTH_SHORT)
+        .thenReturn(toastMock);
+
+    // Act
+    messageListContainerFragment.showToastMessage(TOAST_MESSAGE);
+
+    // Assert
+    Mockito.verify(toastMock).show();
+  }
+
+  @Test
+  public void testSetupListInMessageListingView() {
+    // Arrange
+    List<BaseListItem> baseListItems = new ArrayList<>();
+    PowerMockito.when(messageListContainerFragment.hasPageLoaded()).thenReturn(true);
+    MessageListContract.ConfigureView messageListViewMock = Mockito.mock(MessageListContract.ConfigureView.class);
+    MessageListContainerContract.Presenter presenterMock = Mockito.mock(MessageListContainerContract.Presenter.class);
+    Whitebox.setInternalState(messageListContainerFragment, "mMessageListView", messageListViewMock);
```

Comment:
⚠️  This snippet of code uses so-called magic strings and sometimes is difficult to infer their goal in the application. It is a good practice to "extract" magic strings and provide meaningful names for them. In this case, consider extracting them into class-level symbolic constants. This approach increases maintainability (single point of modification) and debugging, avoid duplication and decrease verbosity.  

**[More details](https://confluence.devfactory.com/x/SI8-Ew)**
          
## The solution in this case

```Java  
    private static final String M_MESSAGE_LIST_VIEW = "mMessageListView";
    // replace 5 usages of this literal with the new constant
```
 

<a id="8ef30fe0-4f1d-11e8-b712-a55fdd633c3d" href="none" codereview-uuid-reserved-tag />